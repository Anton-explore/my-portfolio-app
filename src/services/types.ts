

export type Ref = HTMLElement;

export interface UniProps {
    id: string
}

export interface BoxProps {
    title: string,
    content: string,
    color?: string,
    id?: string
};

export interface ButtonProps {
    text?: string,
    icon?: string | JSX.Element,
    onClick?: () => void;
    disabled?: boolean;
    shrunk?: boolean;
    up?: boolean;
}

export interface ExperienceData {
    date: string;
    info: {
        company: string;
        job: string;
        description: string;
    }
}

export interface ExperienceDataProps {
    data: ExperienceData[];
}

export interface FeedProps {
    feedback: string,
    reporter: {
        photoUrl: string;
        name: string;
        profession: string;
        citeUrl: string
    }
}

export interface FeedDataProps {
    data: FeedProps[];
}

export type InfoProps = {
    text: string;
}

export interface PhotoBoxProps {
    name?: string;
    title?: string;
    description?: string | JSX.Element;
    avatar: string;
    isInline?: boolean;
    facade?: boolean;
}

export type PortfolioInfoProps = {
    title: string,
    text: string,
    url: string
}



export interface MyFormValues extends Record<string, {}> {
  skillName: string,
  skillRange: string | number,
}

export interface TimelineProps {
    id: number,
    date: string,
    title: string,
    description: string
}

export interface TimelineDataProps {
    educations: TimelineProps[];
}


export interface EduStateType {
    educations: TimelineProps[];
    isLoading: boolean,
    error: string | null,
}

export interface SkillsState {
    id: string,
    skillName: string,
    skillRange: string | number,
};

export interface SkillStateType {
    skills: SkillsState[];
    isLoading: boolean,
    error: string | null,
}

