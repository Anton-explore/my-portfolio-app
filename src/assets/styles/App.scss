
.app-wrapper {
    display: flex;
    gap: 10px;
    padding: 0;
}

section {
    margin: 10px;
    padding: 20px 0;
}

li {
    list-style-type: none;
}

a {
    text-decoration: none;
}

p {
    font-size: 14px;
}

.app-main-inner {
    width: 100%;
    padding: 30px;
    padding-left: 0;
    display: flex;
    flex-direction: column;
    justify-content: center;
}


