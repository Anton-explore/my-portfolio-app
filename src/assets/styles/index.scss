// @import-normalize;

body {
  width: 100%;
  max-width: 100%;
  margin: 0;
  font-family: "Open Sans", -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

code {
  font-family: source-code-pro, "Courier New",
    monospace;
}
