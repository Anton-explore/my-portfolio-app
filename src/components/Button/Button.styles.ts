import styled, { css } from "styled-components"

type StyledButtonProps = {
  up?: boolean
}

type StyledHeadingProps = {
  shrunk?: boolean
}

export const StyledButton = styled.button<StyledButtonProps>`
  max-width: 140px;
  border-radius: 5px;
  padding: 10px 20px;
  margin: 10px;
  display: flex;
  justify-content: center;
  gap: 10px;

  background-color: ${({theme}) => theme.colors.oxford};
  color: ${({theme}) => theme.colors.white};
  ${({ up }) =>
    up &&
    css`
      position: fixed;
      bottom: 0px;
      right: 50px;
      z-index: 5;
    `}
    
  &:active,
  &:focus,
  &:hover {
    background-color: ${({theme}) => theme.colors.accent};
    color: ${({theme}) => theme.colors.white};
    cursor: pointer;
  }

  &:disabled {
    cursor: pointer;
    background-color: ${({theme}) => theme.colors.alto};
    box-shadow: none;

    &:hover,
    &:focus {
      cursor: not-allowed;
    }
  }
`

export const StyledHeading = styled.div<StyledHeadingProps>`
    @media (max-width: 1070px) {
      display: ${({ shrunk }) => (shrunk ? 'none' : 'block')};
    }
`
