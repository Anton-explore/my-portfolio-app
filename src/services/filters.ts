import card1 from '../assets/images/card1.jpg';
import card3 from '../assets/images/card3.png';



export const filtersDefault = [
  { "label": "All", active: true },
  { "label": "Ui", active: false },
  { "label": "Code", active: false },
];

export const portfolioItemsDefault = [
  {
    "id": "a",
    "filter": "Ui",
    "imgUrl": card1,
    alt: "Avatar",
    title: "Some text",
    text: "Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo",
    url: "https://somesite.com"
  },
  {
    "id": "b",
    "filter": "Code",
    "imgUrl": card3,
    alt: "Avatar",
    title: "Some text",
    text: "Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo",
    url: "https://somesite.com"
  },
  {
    "id": "c",
    "filter": "Ui",
    "imgUrl": card1,
    alt: "Avatar",
    title: "Some text",
    text: "Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo",
    url: "https://somesite.com"
  },
  {
    "id": "d",
    "filter": "Code",
    "imgUrl": card3,
    alt: "Avatar",
    title: "Some text",
    text: "Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo",
    url: "https://somesite.com"
  },
];