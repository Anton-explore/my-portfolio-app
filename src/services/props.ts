import userAv from '../assets/images/User_avatar_Anton.png';
import card1 from '../assets/images/card1.jpg';
import card3 from '../assets/images/card3.png';
import phone from '../assets/images/phone.svg';
import mail from '../assets/images/mail.svg';
import facebook from '../assets/images/facebook.svg';
import twitter from '../assets/images/twitter.svg';
import skype from '../assets/images/skype.svg';

import { ExperienceDataProps, FeedDataProps, PhotoBoxProps, TimelineDataProps } from './types';

export const BUTTONS_TEXT = {
    BACK: 'Go back',
    RETURN: 'Back to our site',
    SKILL: 'Add skill',
    MORE: 'Know more',
    
}

export const TITLES = {
    ABOUT: 'About me',
    EDU: 'Education',
    EXP: 'Experience',
    SK: 'Skills',
    PF: 'Portfolio',
    CONT: 'Contacts',
    FEED: 'Feedbacks'
}

export const SKILL_LEVEL = {
    BG: 'Beginner',
    PR: 'Proficient',
    EX: 'Expert',
    MS: 'Master'
} 

export const CONTACTS = {
    tel: {
        link: '+500 342 242',
        text: '500 342 242',
        img: phone 
    },
    mail: {
        link: 'famtripuae19@gmail.com',
        text: 'famtripuae19@gmail.com',
        img: mail 
    },
    skype: {
        link: 'kamsolutions.pl',
        text: 'Skype',
        img: skype 
    },
    twitter: {
        link: 'https://twitter.com/wordpress',
        text: 'Twitter',
        img: twitter 
    },
    facebook: {
        link: 'https://www.facebook.com/facebook',
        text: 'Facebook',
        img: facebook 
    },
}

export const nameProps: PhotoBoxProps = {
    name: "Anton Vladykin",
    title: "Front-end developer (Javascript, React)",
    description: "I'm a junior developer. I've switched on this way from studying sociology through the work in the tourism sphere, as a head of the travel agency, gained experience with the demanding clients in the segment of luxury leisure. And for now my goal is to become front end developer. I focus on the efficiency and productivity of my work. I pay great attention to details and strive to achieve a harmony in all of my life. I have improved skills in communication, understanding companions, argumentation and persuasion. I have experience both in team work and self-management.",
    avatar: userAv
}

export const feedbackProps: FeedDataProps = {
    data: [
        {
            feedback: ' Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor',
            reporter: {
                photoUrl: 'https://www.atlassian.com/dam/jcr:ba03a215-2f45-40f5-8540-b2015223c918/Max-R_Headshot%20(1).jpg',
                name: 'John Doe',
                profession: 'Programmer',
                citeUrl: 'https://www.citeexample.com'
            }
        },
        {
            feedback: ' Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor',
            reporter: {
                photoUrl: 'https://www.atlassian.com/dam/jcr:ba03a215-2f45-40f5-8540-b2015223c918/Max-R_Headshot%20(1).jpg',
                name: 'John Doe',
                profession: 'QA',
                citeUrl: 'https://www.citeexample.com'
            }
        }
    ]
};
    
export const timelineProps: TimelineDataProps = {
    educations: [
        {
            id: 1,
            "date": "2023",
            "title": "EPAM UpSkill",
            "description": "Frontend Development. Stack: HTML, CSS, SCSS, JavaScript, Git(Github,Gitlab), Typescript, React, Redux"
        },
        {
            id: 2,
            "date": "2023",
            "title": "DevPRO",
            "description": "Internship Javascript basic, Angular basic, Typescript basic"
        },
        {
            id: 3,
            "date": "2022",
            "title": "QAtestLab",
            "description": "Software Functional Testing (Manual QA). Debugging - creating bug-reports, check-list, check-plan, test-cases.  Game-testing, mobile-testing. Understanding of the life-cycle of Software Testing. SCRUM-methodology."
        },
        {
            id: 4,
            "date": "2022",
            "title": "Prometheus",
            "description": "Basics of Web UI Development.  Javascript basic course. Stack: JavaScript, OOP, HTML, CSS, SCSS, Bootstrap, Git(GitHub), Slack/Jira, Photoshop, Figma"
        
        },
        {
            id: 5,
            "date": "2009",
            "title": "LNU Tarasa Shevchenka",
            "description": "Sociology & PR - master degree, International tourism - practical course"
        }
    ]
}

export const expertiseProps: ExperienceDataProps = {
    data: [
        {
            date: '2022-2023',
            info:
            {
                company: 'EPAM UpSkill',
                job: 'Projects on mentorship EPAM UpSkill',
                description: 'Stack: HTML, CSS, SCSS, JavaScript, Typescript, React, Git(Github,Gitlab). Internship for creating awesome websites and applications'
            }
        },
        {
            date: '2014-2020',
            info:
            {
                company: 'Coral Travel Market LLC',
                job: 'Senior specialist',
                description: 'Working, as a travel agent, and, as a part of department of individual travels, in big international company. I gain experience of communication in international team, and such skills as: systems thinking, tactical and strategic thinking, individual approach, convincing and argumentation'
            }
        },
        {
            date: '2008-2014',
            info:
            {
                company: 'Planeta-Service travel agency',
                job: 'Entrepreneur, Head of the Travel Agency',
                description: 'Working as independent travel agent I increase my income in 200% and received such skills as: listening and attention, problem solving, result-orientation, self-motivation, and experience in decision-making processes.'
            }
        }
    ]
}


export const filtersDefault = [
  { "label": "All", active: true },
  { "label": "Ui", active: false },
  { "label": "Code", active: false },
];

export const portfolioItemsDefault = [
  {
    "id": "a",
    "filter": "Ui",
    "imgUrl": card1,
    alt: "Avatar",
    title: "Some text",
    text: "Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo",
    url: "https://somesite.com"
  },
  {
    "id": "b",
    "filter": "Code",
    "imgUrl": card3,
    alt: "Avatar",
    title: "Some text",
    text: "Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo",
    url: "https://somesite.com"
  },
  {
    "id": "c",
    "filter": "Ui",
    "imgUrl": card1,
    alt: "Avatar",
    title: "Some text",
    text: "Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo",
    url: "https://somesite.com"
  },
  {
    "id": "d",
    "filter": "Code",
    "imgUrl": card3,
    alt: "Avatar",
    title: "Some text",
    text: "Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo",
    url: "https://somesite.com"
  },
];