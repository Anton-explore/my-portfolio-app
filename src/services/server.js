
import { createServer, Model } from "miragejs"

const server = () => createServer({

  models: {
    skill: Model
  },


  routes() {
    this.namespace = "api"


    this.get("/skills", (schema, request) => {
      return schema.skills.all()
    }, { timing: 3000 }
    )


    this.post("/skills", (schema, request) => {
      let attrs = JSON.parse(request.requestBody)
      return schema.skills.create(attrs)
    })
        

    this.delete("/skills/:id", (schema, request) => {
      let id = request.params.id;
      return schema.skills.find(id).destroy()
    })

    this.get("/educations", () => {
        return {
          educations: [
            {
              id: 1,
              "date": "2023",
              "title": "EPAM UpSkill",
              "description": "Frontend Development. Stack: HTML, CSS, SCSS, JavaScript, Git(Github,Gitlab), Typescript, React, Redux"
            },
            {
              id: 2,
              "date": "2023",
              "title": "DevPRO",
              "description": "Internship Javascript basic, Angular basic, Typescript basic"
            },
            {
              id: 3,
              "date": "2022",
              "title": "QAtestLab",
              "description": "Software Functional Testing (Manual QA). Debugging - creating bug-reports, check-list, check-plan, test-cases.  Game-testing, mobile-testing. Understanding of the life-cycle of Software Testing. SCRUM-methodology."
            },
            {
              id: 4,
              "date": "2022",
              "title": "Prometheus",
              "description": "Basics of Web UI Development.  Javascript basic course. Stack: JavaScript, OOP, HTML, CSS, SCSS, Bootstrap, Git(GitHub), Slack/Jira, Photoshop, Figma"
            },
            {
              id: 5,
              "date": "2009",
              "title": "LNU Tarasa Shevchenka",
              "description": "Sociology & PR - master degree, International tourism - practical course"
            }
          ],
        }
      }, { timing: 3000 }
    )

    
  },
})

export default server;