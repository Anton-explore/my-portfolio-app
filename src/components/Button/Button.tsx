import { ButtonProps } from "../../services/types";
import { StyledButton, StyledHeading } from "./Button.styles";


const Button: React.FC<ButtonProps> = ( { text, icon, onClick, disabled, shrunk, up } ) => {

    return (
        <>
            <StyledButton
                type="submit"
                onClick={onClick}
                disabled={!!disabled}
                up={!!up}
            >
                {icon}
                {text ? <StyledHeading shrunk={!!shrunk}>{text}</StyledHeading> : null}
                
            </StyledButton>
        </>
    )

}

export default Button;