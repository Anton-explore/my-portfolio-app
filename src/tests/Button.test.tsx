
import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer'
import { ButtonProps } from '../services/types';
import Button from '../components/Button/Button';
import { ThemeProvider } from 'styled-components';
import { theme } from '../assets/styles/theme';

afterEach(cleanup);

describe('testing Button', () => {

  const defaultProps: ButtonProps = {
    text: 'Test Button',
    onClick: jest.fn(),
  };

  it('snapshot renders button text correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Button {...defaultProps} />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('snapshot renders correctly, falsy values', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <Button />
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls onClick prop when button is clicked', () => {
    render(
      <ThemeProvider theme={theme}>
        <Button {...defaultProps} />
      </ThemeProvider>
    );
    fireEvent.click(screen.getByText('Test Button'));
    expect(defaultProps.onClick).toHaveBeenCalled();
  });

  // it('disables button when disabled prop is true', () => {
  //   render(
  //     <ThemeProvider theme={theme}>
  //       <Button {...defaultProps} disabled />
  //     </ThemeProvider>
  //   );
  //   expect(screen.getByText('Test Button')).toBeDisabled();
  // });
})
