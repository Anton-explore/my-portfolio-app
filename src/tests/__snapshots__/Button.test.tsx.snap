// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Button snapshot renders button text correctly 1`] = `
<button
  className="sc-bcXHqe jsbDui"
  disabled={false}
  onClick={[MockFunction]}
  type="submit"
>
  <div
    className="sc-gswNZR gUeIuu"
  >
    Test Button
  </div>
</button>
`;

exports[`testing Button snapshot renders correctly, falsy values 1`] = `
<button
  className="sc-bcXHqe jsbDui"
  disabled={false}
  type="submit"
/>
`;
